pkg/remote/codec/thrift/binary_protocol.go

```diff
@@ -173,6 +174,9 @@ func (p *BinaryProtocol) WriteBool(value bool) error {
 
 // WriteByte ...
 func (p *BinaryProtocol) WriteByte(value int8) error {
+	if b, ok := p.trans.(remote.FrameWrite); ok {
+		return b.WriteData([]byte{byte(value)})
```

Comment:
This is due to the fact that the buffer doesn't implement malloc, but I don't think this implementation is good.